### PR TITLE
refactor(cmd/artifact): share flags and cleanup descriptions

### PR DESCRIPTION
**Pull request by maxgio92**
**What type of PR is this?**

This PR adds a tiny refactoring and cleanup on `artifact install` and `artifact follow` commands.
In details, adds sharing of flag names as constants between packages and cleans up unneeded commands' description.

/kind cleanup

**Any specific area of the project related to this PR?**

/area cli

**What this PR does / why we need it**:

To remove duplicated code and improve maintainability.
To cleanup commands' documentation.

**Which issue(s) this PR fixes**:

NA

**Special notes for your reviewer**:

NA

### CODE REVIEW

File: cmd/artifact/install/install.go [hunk 1]
```diff
@@ -88,61 +88,65 @@ func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cob
...
 			if f == nil {
 				// should never happen
-				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag rulesfiles-dir"))
+				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag %s", FlagRulesFilesDir))
 			} else if !f.Changed && viper.IsSet(config.ArtifactInstallRulesfilesDirKey) {
+
 				val := viper.Get(config.ArtifactInstallRulesfilesDirKey)
 				if err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val)); err != nil {
-					o.Printer.CheckErr(fmt.Errorf("unable to overwrite \"rulesfiles-dir\" flag: %w", err))
+					o.Printer.CheckErr(fmt.Errorf("unable to overwrite \"%s\" flag: %w", FlagRulesFilesDir, err))
```
Comment (alacuku) [inline cmd/artifact/install/install.go:101]:
```suggestion
					o.Printer.CheckErr(fmt.Errorf("unable to overwrite %q flag: %w", FlagRulesFilesDir, err))
```

File: cmd/artifact/install/install.go [hunk 2]
```diff
@@ -88,61 +88,65 @@ func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cob
...
 				}
 			}
-			// Check if directory exists and is writable
+
+			// Check if directory exists and is writable.
 			if err := utils.ExistsAndIsWritable(f.Value.String()); err != nil {
-				o.Printer.CheckErr(fmt.Errorf("rulesfiles-dir: %w", err))
+				o.Printer.CheckErr(fmt.Errorf("%s: %w", FlagRulesFilesDir, err))
```
Comment (alacuku) [inline cmd/artifact/install/install.go:107]:
```suggestion
				o.Printer.CheckErr(fmt.Errorf("%q: %w", FlagRulesFilesDir, err))
```

File: cmd/artifact/install/install.go [hunk 3]
```diff
@@ -88,61 +88,65 @@ func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cob
...
 			}
 
 			// Override "plugins-dir" flag with viper config if not set by user.
-			f = cmd.Flags().Lookup("plugins-dir")
+			f = cmd.Flags().Lookup(FlagPluginsFilesDir)
 			if f == nil {
 				// should never happen
-				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag plugins-dir"))
+				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag %s", FlagPluginsFilesDir))
```
Comment (alacuku) [inline cmd/artifact/install/install.go:114]:
```suggestion
				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag %q", FlagPluginsFilesDir))
```

File: cmd/artifact/install/install.go [hunk 4]
```diff
@@ -88,61 +88,65 @@ func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cob
...
 			} else if !f.Changed && viper.IsSet(config.ArtifactInstallPluginsDirKey) {
 				val := viper.Get(config.ArtifactInstallPluginsDirKey)
 				if err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val)); err != nil {
-					o.Printer.CheckErr(fmt.Errorf("unable to overwrite \"plugins-dir\" flag: %w", err))
+					o.Printer.CheckErr(fmt.Errorf("unable to overwrite \"%s\" flag: %w", FlagPluginsFilesDir, err))
```
Comment (alacuku) [inline cmd/artifact/install/install.go:118]:
```suggestion
					o.Printer.CheckErr(fmt.Errorf("unable to overwrite %q flag: %w", FlagPluginsFilesDir, err))
```

File: cmd/artifact/install/install.go [hunk 5]
```diff
@@ -88,61 +88,65 @@ func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cob
...
 				}
 			}
-			// Check if directory exists and is writable
+
+			// Check if directory exists and is writable.
 			if err := utils.ExistsAndIsWritable(f.Value.String()); err != nil {
-				o.Printer.CheckErr(fmt.Errorf("plugins-dir: %w", err))
+				o.Printer.CheckErr(fmt.Errorf("%s: %w", FlagPluginsFilesDir, err))
```
Comment (alacuku) [inline cmd/artifact/install/install.go:124]:
```suggestion
				o.Printer.CheckErr(fmt.Errorf("%q: %w", FlagPluginsFilesDir, err))
```

File: cmd/artifact/install/install.go [hunk 6]
```diff
@@ -88,61 +88,65 @@ func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cob
...
 			}
 
 			// Override "allowed-types" flag with viper config if not set by user.
-			f = cmd.Flags().Lookup("allowed-types")
+			f = cmd.Flags().Lookup(FlagAllowedTypes)
 			if f == nil {
 				// should never happen
-				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag allowed-types"))
+				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag %s", FlagAllowedTypes))
```
Comment (alacuku) [inline cmd/artifact/install/install.go:131]:
```suggestion
				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag %q", FlagAllowedTypes))
```

File: cmd/artifact/install/install.go [hunk 7]
```diff
@@ -88,61 +88,65 @@ func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cob
...
 			} else if !f.Changed && viper.IsSet(config.ArtifactAllowedTypesKey) {
 				val, err := config.ArtifactAllowedTypes()
 				if err != nil {
 					o.Printer.CheckErr(err)
 				}
 				if err := cmd.Flags().Set(f.Name, val.String()); err != nil {
-					o.Printer.CheckErr(fmt.Errorf("unable to overwrite \"allowed-types\" flag: %w", err))
+					o.Printer.CheckErr(fmt.Errorf("unable to overwrite \"%s\" flag: %w", FlagAllowedTypes, err))
```
Comment (alacuku) [inline cmd/artifact/install/install.go:138]:
```suggestion
					o.Printer.CheckErr(fmt.Errorf("unable to overwrite %s flag: %w", FlagAllowedTypes, err))
```

File: cmd/artifact/install/install.go [hunk 8]
```diff
@@ -88,61 +88,65 @@ func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cob
...
 				}
 			}
 
-			f = cmd.Flags().Lookup("resolve-deps")
+			f = cmd.Flags().Lookup(FlagResolveDeps)
 			if f == nil {
 				// should never happen
-				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag resolve-deps"))
+				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag %s", FlagResolveDeps))
```
Comment (alacuku) [inline cmd/artifact/install/install.go:145]:
```suggestion
				o.Printer.CheckErr(fmt.Errorf("unable to retrieve flag %q", FlagResolveDeps))
```

File: cmd/artifact/install/install.go [hunk 9]
```diff
@@ -88,61 +88,65 @@ func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cob
...
 			} else if !f.Changed && viper.IsSet(config.ArtifactInstallResolveDepsKey) {
 				val := viper.Get(config.ArtifactInstallResolveDepsKey)
 				if err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val)); err != nil {
-					o.Printer.CheckErr(fmt.Errorf("unable to overwrite \"resolve-deps\" flag: %w", err))
+					o.Printer.CheckErr(fmt.Errorf("unable to overwrite \"%s\" flag: %w", FlagResolveDeps, err))
```
Comment (alacuku) [inline cmd/artifact/install/install.go:149]:
```suggestion
					o.Printer.CheckErr(fmt.Errorf("unable to overwrite %q flag: %w", FlagResolveDeps, err))
```

**Comment by maxgio92** [comment]
Thank you @alacuku

**Comment by maxgio92** [comment]
Suggestions applied @alacuku, thank you.

**Review by alacuku — APPROVED** [review]

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**